components/support/base/src/main/java/mozilla/components/support/base/observer/ObserverRegistry.kt

```diff
@@ -117,6 +130,7 @@ class ObserverRegistry<T> : Observable<T> {
     /**
      * GenericLifecycleObserver implementation to bind an observer to a Lifecycle.
      */
+    @SuppressLint("RestrictedApi")
```

Comment:
Is this suppress needed? I can't replicate the lint error.

Reply:
Probably yes. :) `GenericLifecycleObserver` (the one we are extending here) has restricted scope [LIBRARY](https://android.googlesource.com/platform/frameworks/support/+/refs/heads/master/lifecycle/common/src/main/java/androidx/lifecycle/GenericLifecycleObserver.java#25).

Reply:
We should not suppress this and fix that - Let's file a follow-up issue. :)

Reply:
Follow-up: #3140

Reply:
@pocmo @NotWoods Thanks. :+1: